Source/Microsoft.Teams.Apps.CompanyCommunicator/Authentication/AuthenticationServiceCollectionExtensions.cs

```diff
@@ -135,5 +136,39 @@ private static void RegisterAuthorizationPolicy(IServiceCollection services)
 
             services.AddSingleton<IAuthorizationHandler, MustBeValidUpnHandler>();
         }
+
+        private static bool AudienceValidator(
+            IEnumerable<string> audiences,
+            SecurityToken securityToken,
+            TokenValidationParameters validationParameters)
+        {
+            if (audiences == null || audiences.Count() == 0)
+            {
+                throw new ApplicationException("No audience defined in token!");
+            }
+
+            var validAudiences = validationParameters.ValidAudiences;
+            if (validAudiences == null || validAudiences.Count() == 0)
+            {
+                throw new ApplicationException("No valid audience defined in validationParameters!");
+            }
+
+            var validAudienceHashSet = new HashSet<string>(validAudiences, new AudienceComparer());
```

Comment:
I think we can just use Linq and Any() with a case insensitive string comparison so we can remove this HashSet and the private class